nifi-registry-core/nifi-registry-web-api/src/test/resources/keys/README.md

```diff
@@ -12,36 +12,217 @@
+| - | ca-key.pem | CA private (signing) key | PEM | password |
+| - | ca-ts.jks | CA cert truststore (shared by clients and servers) | JKS | password |
+| - | ca-ts.p12 | CA cert truststore (shared by clients and servers) | PKCS12 | password |
+| registry, localhost | registry-cert.pem | NiFi Registry server public cert | PEM (unencrypted) | N/A |
+| registry, localhost | registry-key.pem | NiFi Registry server private key | PEM | password |
+| registry, localhost | registry-ks.jks | NiFi Registry server key/cert keystore | JKS | password |
+| registry, localhost | registry-ks.p12 | NiFi Registry server key/cert keystore | PKCS12 | password |
+| CN=user1, OU=nifi | user1-cert.pem | client (user="user1") public cert | PEM (unencrypted) | N/A |
+| CN=user1, OU=nifi | user1-key.pem | client (user="user1") private key | PEM | password |
+| CN=user1, OU=nifi | user1-ks.jks | client (user="user1") key/cert keystore | JKS | password |
+| CN=user1, OU=nifi | user1-ks.p12 | client (user="user1") key/cert keystore | PKCS12 | password |
 
-    keys/
-     +-- client-ks.jks      # client keystore: keystorePass=clientKeystorePassword, keyPass=u1Pass
-     +-- localhost-ks.jks   # server keystore: keystorePass=localhostKeystorePassword, keyPass=localhostKeystorePassword
-     +-- localhost-ts.jks   # server/client truststore (contains CA): truststorePass=localhostTruststorePassword
+## Generating Additional Test Keys/Certs
 
-Copy these files to the test/resources/keys/ directory.
+If we need to add a service to our test environment that requires a cert signed by the same CA, here are the steps for generating additional keys for thsi directory that are signed by the same CA key.
 
+Requirements:
+
+- docker
+- keytool (included with Java)
+- openssl (included/available on most platforms)
+
+If you do not have docker, you can alternatively use the nifi-toolkit binary which has a dependency on Java 1.8 and is available for download from http://nifi.apache.org. 
+
+### New Service Keys
+
+The steps for generating a new *service* key/cert pair are (using `proxy` as the example service):
+
+```
+# make working directory
+rm -rf /tmp/test-keys
+mkdir /tmp/test-keys
+cd /tmp/test-keys
+
+# copy existing CA key/cert pair to working directory, rename to default tls-toolkit names
+cp /path/to/nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/resources/keys/ca-key.pem ./nifi-key.key
+cp /path/to/nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/resources/keys/ca-cert.pem ./nifi-cert.pem
+
+# use NiFi Toolkit Docker image to generate new keys/certs
+docker run -v /tmp/test-keys:/tmp -w /tmp apache/nifi-toolkit:latest tls-toolkit standalone \
+      --hostnames proxy \
+      --subjectAlternativeNames localhost \
+      --nifiDnSuffix ", OU=nifi" \
+      --keyStorePassword password \
+      --trustStorePassword password \
+      --days 9999 \
+      -O
+
+# switch to output directory, create final output directory
+cd /tmp/test-keys
```

Comment:
Isn't the user still in `/tmp/test-keys` at this point?

Reply:
yeah just had it there in case they did not run these command all together. it should not hurt if they are already there